src/main/java/org/variantsync/diffdetective/validation/EditClassValidation.java

```diff
@@ -0,0 +1,70 @@
+package org.variantsync.diffdetective.validation;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.variantsync.diffdetective.analysis.Analysis;
+import org.variantsync.diffdetective.analysis.FilterAnalysis;
+import org.variantsync.diffdetective.analysis.PreprocessingAnalysis;
+import org.variantsync.diffdetective.analysis.StatisticsAnalysis;
+import org.variantsync.diffdetective.datasets.Repository;
+import org.variantsync.diffdetective.editclass.proposed.ProposedEditClasses;
+import org.variantsync.diffdetective.metadata.EditClassCount;
+import org.variantsync.diffdetective.variation.diff.filter.DiffTreeFilter;
+import org.variantsync.diffdetective.variation.diff.transform.CutNonEditedSubtrees;
+
+/**
+ * This is the validation from our ESEC/FSE'22 paper.
+ * It provides all configuration settings and facilities to setup the validation by
+ * creating a {@link Analysis} and run it.
+ * @author Paul Bittner
+ */
+public class EditClassValidation implements Analysis.Hooks {
+    // This is only needed for the `MarlinDebug` test.
+    public static final BiFunction<Repository, Path, Analysis> AnalysisFactory = (repo, repoOutputDir) -> new Analysis(
+        "EditClassValidation",
+        List.of(
+            new PreprocessingAnalysis(new CutNonEditedSubtrees()),
+            new FilterAnalysis(DiffTreeFilter.notEmpty()), // filters unwanted trees
+            new EditClassValidation(),
+            new StatisticsAnalysis()
+        ),
+        repo,
+        repoOutputDir
+    );
```

Comment:
This new compositional approach is indeed quite nice! Good job!

Reply:
thanks